lib/AST/ASTContext.cpp

```diff
@@ -276,8 +276,12 @@ struct ASTContext::Implementation {
     ~Arena() {
       for (auto &conformance : SpecializedConformances)
         conformance.~SpecializedProtocolConformance();
+      // Work around MSVC warning: local variable is initialized but
+      // not referenced.
+#pragma warning (disable: 4189)
```

Comment:
More unguarded pragmas. Also, I find it hard to believe that we'd get a warning on this code and not the previous line, so maybe this is no longer necessary?